entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy

```diff
@@ -304,13 +420,68 @@ declare function {$prefix}:instance-to-envelope(
                 {$prefix}:instance-to-canonical-xml($entity-instance)
             }},
             element es:attachments {{
-                map:get($entity-instance, '$attachments')
+                for $attachment in $entity-instance=>map:get('$attachments')
+                return
+                    typeswitch ($attachment)
+                    case object-node() return xdmp:quote($attachment)
+                    case array-node() return xdmp:quote($attachment)
+                    default return $attachment
             }}
         }}
     }}
 }};
 
 
+(:
+ : @param $entity-instance an instance, as returned by an extract-instance
+ : function
+ : @return A document which wraps both the canonical instance and source docs.
+ :)
+declare function {$prefix}:instance-to-envelope(
```

Comment:
Do we need this wrapper, now that we have explicit xml and json envelope functions? How or where can this be of use?